### PR TITLE
Don't depend on Cohttp_lwt.Client

### PR DESCRIPTION
**Pull request by rgrinberg**
Create your own client signature that doesn't include more fields than
necessary

@marklrh

### CODE REVIEW

File: lib/oauth.ml [hunk 1]
```diff
@@ -208,11 +208,28 @@ module type OAuth_client = sig
       [> `Ok of (Cohttp.Header.t * string) | `Error of oauth_error] Lwt.t
 end
 
+module type Client = sig
+  val post :
+    ?ctx:Cohttp_lwt_unix.Client.ctx ->
```
Comment (marklrh) [inline lib/oauth.ml]: @rgrinberg um, what if we want to use it on Mirage? is Cohttp_lwt_unix still OK?

**Comment by rgrinberg** [comment]
Oh yeah. The only problem should be the context variable though.

**Comment by rgrinberg** [comment]
@marklrh i fixed it

**Comment by marklrh** [comment]
merged. thanks!

**Pull request merged**